homeassistant/components/lutron_caseta/__init__.py

```diff
@@ -371,7 +371,9 @@ def serial(self):
     @property
     def unique_id(self):
         """Return the unique ID of the device (serial)."""
-        return str(self.serial)
+        if self._device["serial"] is None:
+            return f'{self._bridge_device["serial"]}_{self.device_id}'
+        return str(self._device["serial"])
```

Comment:
Note that serial is an `int` so if we don't `str()` it here it won't be backwards compat

Reply:
yes, ok understood.

I'm working on an alternate fix.  I can add a method _handle_non_serial(self, serial), and do the None check here.  And use this method in the unique_id property return, and the identifiers setup.

Back shortly.